### ISSUE TITLE
Memory leak when connection fails

### ISSUE THREAD

**Issue by gjonikas** [amqp-rs/lapin#368]
Hi,
I believe I've found a memory leak which occurs when lapin fails to establish an amqp connection. 

It is possible to reproduce the issue with the code snippet below. The re is no RabbitMQ running on 127.0.0.1:5672, so the connection fails.  Retrying to connect in a tight loop makes program's RSS memory grow indefinitely.

This happens with the latest Lapin 2.1.1, but I've initially encountered it with 1.x version.

```
fn main() {
    futures::executor::LocalPool::new().run_until(async {
        loop {
            let conn = lapin::Connection::connect(
                "amqp://127.0.0.1:5672/%2F?heartbeat=5&connection_timeout=10000",
                lapin::ConnectionProperties::default(),
            )
            .await;
            println!("connection {:#?}", conn);
        }
    });
}
```

Please let me know if there is any other information I could provide you with.

**Comment by gjonikas** [comment]
Hi @dmgolembiowski, this was built using "amd64/rust:1.61.0-bullseye" docker image. The binary was later run within "amd64/debian:bullseye-slim" docker.

But I was also able to recreate the issue on my Fedora machine using the example given above with the latest rustc that was available back when this was reported (1.62?).

I've not tried using the backoff crate. My workaround for this issue was increasing the retry timeout to a high enough value so we are very unlikely to get OOM without noticing the connectivity issues first.

This is acceptable in my scenario, because my use case can tolerate couple mins of delay of those messages.

I hope this helps.

**Comment by Keruspe** [comment]
Hi,

I'll give it a look thanks
